assets/aml-benchmark/components/src/inference_postprocessor/inference_postprocessor.py

```diff
@@ -90,129 +134,606 @@ def __post_init__(self) -> None:
+                        curr_pred_list = []
+                        for i in range(0, len(predicted)):
+                            curr_pred = {self.prediction_column_name: predicted[i]}
+                            out_string = jinja_template.render(curr_pred)
+                            curr_pred_list.append(out_string)
+                        pred_list.append(curr_pred_list)
+                    else:
+                        raise BenchmarkUserException._with_error(
+                            AzureMLError.create(BenchmarkUserError, error_details=e)
+                        )
+            else:
+                out_string = jinja_template.render(row)
+                pred_list.append(out_string)
+        if isinstance(pred_list[0], list) and len(pred_list[0]) > 1:
+            cols = [
+                f"{self.prediction_column_name}_{i+1}" for i in range(len(pred_list[0]))
+            ]
+        else:
+            cols = self.prediction_column_name
+        result_df[cols] = pred_list
+        # combine the records in one pandas dataframe and write it to the jsonl file.
+        result_df.to_json(self.result, lines=True, orient="records")
         return
 
-    def run(self) -> None:
+    def apply_generic_processor(self, out_string: str, row: dict = None) -> List:
+        """Processor steps."""
+        out_string = self.apply_remove_prompt_prefix(out_string, row)
+        out_string = self.apply_remove_prefixes(out_string)
+        out_string = self.apply_separator(out_string)
+        out_string = self.apply_find_first(out_string)
+        out_string = self.apply_extract_number(out_string)
+        out_string = self.apply_regex_expr(out_string)
+        out_string = self.apply_strip_characters(out_string)
+        out_string = self.apply_label_map(out_string)
+        return out_string
+
+    def extract_inferences(
+        self, result_df: pd.DataFrame, key: str = None, processor_order: List = None
+    ) -> pd.DataFrame:
+        """Extract inferences using generic method if no template or custom post-processor is provided."""
+        predicted_data = read_jsonl_files(resolve_io_path(self.prediction_dataset))
+        pred_list = []
+        if self.prediction_column_name in predicted_data[0].keys():
+            key = self.prediction_column_name
+        else:
+            key = key if key else "0"
+        for row in predicted_data:
+            predicted = row.get(key)
+            if isinstance(predicted, list) and len(predicted[0]) > 1:
+                curr_pred_list = []
+                for i in range(0, len(predicted)):
+                    out_string = predicted[i]
+                    out_string = self.apply_generic_processor(out_string, row)
+                    curr_pred_list.append(out_string)
+                pred_list.append(curr_pred_list)
+            else:
+                out_string = predicted if isinstance(predicted, str) else predicted[0]

```

Comment:
Is it safe to assume that `predicted[0]` won't throw an error here? If `predicted` is an empty array, should we handle it by using a default value like empty string or throw a proper error message?

Reply:
@arun-rajora I have never come across any scenario where Null was predicted. Either there would be a text or empty string. And in case of empty strings, it has worked before. Theoretically, I don't think there could be a scenario where nothing gets predicted as we never pass Null to the model.

Reply:
I can imagine when working with endpoints, these assumptions can easily break where the API requests to the endpoints can fail as well. So, it is better if we handle very minor corner cases as well.